components/support/error/src/macros.rs

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/// Tell the application to report an error
+///
+/// This will cause the error to be reporting to a Sentry-like system.  We will send extra context
+/// for the error, including a backtrace.  This should only be used for errors that we don't expect
+/// to see and will work on fixing if we see a non-trivial volume of them.
+///
+/// type_name identifies the error.  It should be the main text that gets shown to the
+/// user and also how the error system groups errors together.  It should be in UpperCamelCase
+/// form.
+///
+/// Good type_names require some trial and error, for example:
+///   - Start with the error kind variant name
+///   - Add more text to distinguish errors more.  For example an error code, or an extra word
+///     based on inspecting the error details
+#[macro_export]
+macro_rules! report_error {
+    ($type_name:expr, $($arg:tt)*) => {
+        let message = std::format!($($arg)*);
+        ::log::warn!("report {}: {}", $type_name, message);
+        $crate::report_error($type_name.to_string(), message.to_string());
+    };
+}
+
+/// Log a breadcrumb if we see an `Result::Err` value
+///
+/// Use this macro to wrap a function call that returns a `Result<>`.  If that call returns an
+/// error, then we will log a breadcrumb for it.  This can be used to track down the codepath where
+/// an error happened.
+#[macro_export]
+macro_rules! trace_error {
+    ($result:expr) => {{
```

Comment:
ISTM that in some complex scenarios, we may not be 100% sure which `trace_error!` call caused this? ie, is it worth adding additional context so we know for sure which function caused it?

Reply:
`breadcrumb!` includes the module, line, and column for this very reason.